src/empty-state.tsx

```diff
@@ -0,0 +1,169 @@
+        flex: 1,
+    },
+    largeImage: {
+        width: '100%',
+        aspectRatio: '16/9',
+    },
+    smallImage: {
+        height: 128,
+        [theme.mq.tabletOrSmaller]: {
+            height: 112,
+        },
+    },
+}));
+
+interface BaseProps {
+    title: string;
+    button?: React.ReactElement<typeof ButtonPrimary> | React.ReactElement<typeof ButtonSecondary>;
+    buttonLink?: React.ReactElement<typeof ButtonLink>;
+    description?: string;
+    children?: void;
+    'aria-label'?: string;
+    /** "data-" prefix is automatically added. For example, use "testid" instead of "data-testid" */
+    dataAttributes?: DataAttributes;
+}
+
+interface ImageProps extends BaseProps {
+    imageUrl: string;
+
+    largeImageUrl?: undefined;
+    icon?: undefined;
+}
+
+interface LargeImageProps extends BaseProps {
+    largeImageUrl: string;
+
+    imageUrl?: undefined;
+    icon?: undefined;
+}
+
+interface IconProps extends BaseProps {
+    icon: React.ReactElement<any>;
+
+    imageUrl?: undefined;
+    largeImageUrl?: undefined;
+}
+
+type Props = IconProps | ImageProps | LargeImageProps;
+
+const EmptyState: React.FC<Props> = ({
+    title,
+    description,
+    button,
+    buttonLink,
+    largeImageUrl,
+    imageUrl,
+    icon,
+    'aria-label': ariaLabel,
+    dataAttributes,
+}) => {
+    const {colors} = useTheme();
+    const needsButtonLinkAlignment = buttonLink && !button;
+    const classes = useStyles({largeImageUrl, needsButtonLinkAlignment});
+    const {isTabletOrSmaller} = useScreenSize();
+
+    let resizedIcon;
+    if (icon) {
+        resizedIcon = React.cloneElement(icon, {size: isTabletOrSmaller ? 64 : 80, color: colors.brand});
+    }
+
+    let largeImage, image;
+    if (largeImageUrl) {
+        largeImage = <img className={classes.largeImage} alt="" src={largeImageUrl} />;
+    }
+    if (imageUrl) {
+        image = <img className={classes.smallImage} alt="" src={imageUrl} />;
+    }
+
+    if (isTabletOrSmaller) {
+        return (
+            <section
+                className={classes.container}
+                aria-label={ariaLabel}
+                {...getPrefixedDataAttributes(dataAttributes)}
+            >
+                <Stack space={24}>
+                    {largeImage ?? image ?? resizedIcon}
+                    <Stack space={16}>
+                        <Text4 regular>{title}</Text4>
+                        <Text2 regular color={colors.textSecondary}>
+                            {description}
+                        </Text2>
+                    </Stack>
+                    {button && (
+                        <Inline space={16} alignItems="center">
+                            {button}
+                            {buttonLink}
+                        </Inline>
+                    )}
+                </Stack>
+            </section>
+        );
+    }
+
+    return (
+        <ResponsiveLayout dataAttributes={dataAttributes}>
+            <Boxed>
```

Comment:
This component shouldn't have a ResponsiveLayout. The layout should be given by the feature that uses this component.